### DISCUSSION TITLE
Handling keywords in metadata import to Invenio

### DISCUSSION THREAD

**Discussion by monotasker** [MESH-Research/knowledge-commons-repository#73] [General]
Hi folks. I think I know the answer to this, but it's good to have a record of it here: How are we going to store the user-defined keywords in Invenio records? 

Invenio treats these as one subject "scheme" (something like "user_keywords") and includes them in the large "subjects" field. Each "subject" is a set of key-value pairs (dict or object) that includes not just the subject term, but also the scheme to which the subject belongs. So it's easy to separate, say, "fast" terms from "user_keywords" based on scheme.

But what I've heard so far is that we'd like better separation of the controlled vocabulary subjects from user-defined keywords in the metadata. Since there isn't anything for keywords like this in Invenio or Datacite, I'll create a custom field like "kcr:user_defined_keywords". Am I right that this is the way we'd like to go?

**Comment by bjr70** [comment]
Yes, this is the way I think we should go. We want tags strictly separated from subjects.

**Comment by monotasker** [comment]
Mike agrees strongly. So I'll mark this resolved.
